tests/rados/test_fast_ec_config_params.py

```diff
@@ -935,6 +944,659 @@ def run(ceph_cluster, **kw):
+                log.info("  - object_size: %d bytes", object_size_bytes)
+                log.info("  - Expected savings: %.2f%%", expected_savings)
+                log.info(
+                    "  - Minimum threshold (with %.0f%% leeway): %.2f%%",
+                    savings_tolerance * 100,
+                    min_savings_threshold,
+                )
+
+                if space_saved_percent < min_savings_threshold:
+                    log.error(
+                        "Space savings verification FAILED: %.2f%% (threshold: %.2f%%)",
+                        space_saved_percent,
+                        min_savings_threshold,
+                    )
+                    log.error(
+                        "Pool1 (without optimizations) USED: %s bytes",
+                        pool1_bytes_used,
+                    )
+                    log.error(
+                        "Pool2 (with optimizations) USED: %s bytes",
+                        pool2_bytes_used,
+                    )
+                    raise Exception(
+                        f"Space savings verification failed: {space_saved_percent:.2f}% "
+                        f"is below minimum threshold of {min_savings_threshold}%"
+                    )
+
+                log.info(
+                    "SUCCESS: Pool with EC optimizations uses %s less bytes (%.2f%% savings)",
+                    space_saved,
+                    space_saved_percent,
+                )
+                log.info(
+                    "This confirms that zeros are NOT appended to objects smaller than "
+                    "stripe width when EC optimizations are enabled"
+                )
+            else:
+                log.error(
+                    "Pool stats show zero bytes_used - pool1: %s, pool2: %s",
+                    pool1_bytes_used,
+                    pool2_bytes_used,
+                )
+                raise Exception("Invalid pool stats - bytes_used is zero")
+
+            # Step 7: Verify shard sizes using ceph-objectstore-tool
+            # This provides direct evidence of zero-padding at the OSD shard level
+            log.info("=" * 80)
+            log.info("Step 7: Verifying shard sizes using ceph-objectstore-tool")
+            log.info("=" * 80)
+
+            shard_sizes = {}
+
+            for pool_name in [pool_name_1, pool_name_2]:
+                log.info("Checking shard size for pool: %s", pool_name)
+
+                # Get list of objects in the pool using run_ceph_command (returns JSON)
+                obj_list = rados_obj.run_ceph_command(cmd=f"rados -p {pool_name} ls")
+                objects = [obj["name"] for obj in obj_list] if obj_list else []
+
+                if not objects:
+                    log.warning("No objects found in pool %s", pool_name)
+                    continue
+
+                # Select a random object for verification
+                test_obj = random.choice(objects)
```

Comment:
By using "rados -p <pool_name> ls | shuf -n 1"  we can get an random object.

Reply:
I prefer the current implementation. let me know if the changes are mandatory.